evalml/pipelines/components/transformers/preprocessing/decomposer.py

```diff
@@ -151,7 +151,9 @@ def determine_periodicity(
                 period is detected, returns None.
 
         """
-        X, y = cls._handle_nullable_types(cls, X, y)
+        # Only need to handle nullable types on pandas < 2. Kept for backwards compatibility with pandas 1.x.
+        if pd.__version__[0] == "1":
```

Comment:
Is this really the best way to check the version? It looks janky and brittle

Reply:
Yeah I'm not a huge fan of this too to be honest. [This](https://stackoverflow.com/questions/20612645/how-to-find-the-installed-pandas-version) made it seem like this was the best option, given that `pd.__version__` returns the version number as a string. Less pandas-specific resources [here](https://stackoverflow.com/questions/710609/checking-a-python-module-version-at-runtime) still return the version as as string. I am very much open to alternative implementations/suggestions if anyone has any!

Reply:
Would it make more sense to do something like `int(pd.__version__.split('.')[0]) < 2`? I highly doubt this code would be here if pandas ever got to a version 10.x, but it feels a little safer than string comparison?

Reply:
Good call, will replace.